fish-rust/src/env_dispatch.rs

```diff
@@ -693,11 +695,14 @@ fn init_locale(vars: &EnvStack) {
         "C.UTF-8", "en_US.UTF-8", "en_GB.UTF-8", "de_DE.UTF-8", "C.utf8", "UTF-8",
     ];
 
-    let old_msg_locale: CString = unsafe {
-        let old = libc::setlocale(libc::LC_MESSAGES, std::ptr::null());
+    let old_msg_locale: CString = {
+        let old = unsafe { libc::setlocale(libc::LC_MESSAGES, ptr::null()) };
+        assert_ne!(old, ptr::null_mut());
```

Comment:
```suggestion
        assert!(!old.is_null());
```

Reply:
It was intentional to have it more verbose, `assert!(!` is a bit syntax-heavy, but there is no `is_not_null`